### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by tomszosz**
initial release

### CODE REVIEW

File: conf/limited.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+# optional `language` parameter, default en
```
Comment (arkadiuszbeer) [inline conf/limited.yaml:2]: Remove this comment

**Review by arkadiuszbeer — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -1,2 +1,162 @@
...
+
+    _Description_: Simple configuration data directly in yaml format
+
+    _Required_: `false`
+
+    _Default_: ""
+
+* `format-output`:
+
+    _Description_: Format of output
```
Comment (KlaudiaBB) [inline README.md]:
```suggestion
    _Description_: Format of an output
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,162 @@
...
+
+    _Description_: Upload result as an artifact
+
+    _Required_: `false`
+
+    _Default_: "true"
+
+* `presidio-cli-version`:
+
+    _Description_: Presidio cli version - version of presidio-cli used in action
```
Comment (KlaudiaBB) [inline README.md]:
```suggestion
    _Description_: Presidio cli version - a version of presidio-cli used in action.
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,162 @@
...
+
+    _Required_: `false`
+
+    _Default_: "1.0.0"
+
+## Outputs
+
+Output depends from `format-output` parameter:
```
Comment (KlaudiaBB) [inline README.md]:
```suggestion
An output depends on the `format-output` parameter:
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,162 @@
...
+
+* auto - default format, switches automatically between those 2 modes:
+  * github, if run on github - environment variables `GITHUB_ACTIONS` and `GITHUB_WORKFLOW` are set
+  * colored, otherwise
+
+## How it works
+
+Presidio action uses [presidio-cli](https://pypi.org/project/presidio-cli/)
+based on presidio-analyzer from [Microsoft Presidio framework](https://github.com/microsoft/presidio)
+to check code agains unwanted types of data like 'EMAIL_ADDRESS', 'PHONE_NUMBER' inside application code.
```
Comment (KlaudiaBB) [inline README.md]:
```suggestion
to check code against undesirable types of data such as 'EMAIL_ADDRESS' or 'PHONE_NUMBER' inside application's code.
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,162 @@
...
+
+Full list of [supported entities](https://microsoft.github.io/presidio/supported_entities/)
```
Comment (KlaudiaBB) [inline README.md]:
```suggestion
For more information please see a full [list of supported entities](https://microsoft.github.io/presidio/supported_entities/).
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,162 @@
...
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Produce the presidio report
+        uses: insightsengineering/presidio-action@v1
+        # all parameters below are optional
+        with:
+          # path to project.
+          # if project cannot is in specific 'my-project' path,
```
Comment (KlaudiaBB) [inline README.md]:
```suggestion
          # if project does not have a specific 'my-project' path,
```

File: action.yaml [hunk 8]
```diff
@@ -0,0 +1,96 @@
+---
+name: "Presidio client - execute analyze action"
+description: "Github Action to execute Presidio framework using presidio_cli"
```
Comment (cicdguy) [inline action.yaml]:
```suggestion
description: "Github Action to execute PII checks using the Presidio CLI"
```

File: action.yaml [hunk 9]
```diff
@@ -0,0 +1,96 @@
+---
+name: "Presidio client - execute analyze action"
```
Comment (cicdguy) [inline action.yaml]:
```suggestion
name: "Presidio Action"
```

File: action.yaml [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+description: "Github Action to execute Presidio framework using presidio_cli"
+inputs:
+  verify-path:
```
Comment (cicdguy) [inline action.yaml]: Might as well call it `path`. Keep it simple since this is an exclusive action.

File: action.yaml [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    default: "."
+  configuration-file:
+    description: Path to configuration file or predefined configuration (default, limited)
+    required: false
+    default: "default"
+  configuration-data:
+    description: Simple configuration data directly
+    required: false
+    default: ""
+  format-output:
```
Comment (cicdguy) [inline action.yaml]: Call it `output`

File: action.yaml [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+  publish:
+    description: Publish result in PR comment
+    required: false
+    default: "true"
+  upload:
+    description: Uploads as artifact
+    required: false
+    default: "true"
+  presidio-cli-version:
+    description: Presidio cli version
```
Comment (cicdguy) [inline action.yaml]:
```suggestion
    description: Presidio CLI version
```

Fun fact - CLI is not _client_, but _Command Line Interface_

File: action.yaml [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+
+    - name: Get branch names
+      id: branch-names
+      uses: tj-actions/branch-names@v5.1
+
+    - name: Install en language models
+      run: |
+        python -m spacy download en_core_web_sm
+        python -m spacy download en_core_web_lg
+      shell: bash
```
Comment (cicdguy) [inline action.yaml]: The action should allow for other models to be installed via an input.

File: README.md [hunk 14]
```diff
@@ -1,2 +1,162 @@
 # presidio-action
```
Comment (cicdguy) [inline README.md]:
```suggestion
# Presidio Action
```

File: .github/workflows/test-action.yaml [hunk 15]
```diff
@@ -0,0 +1,77 @@
+---
+name: Presidio-cli action test 🧪
```
Comment (cicdguy) [inline .github/workflows/test-action.yaml]:
```suggestion
name: Test 🧪
```

File: action.yaml (hunk 10 above)
Comment (tomszosz) [reply action.yaml]: changed

File: action.yaml (hunk 11 above)
Comment (tomszosz) [reply action.yaml]: changed

File: action.yaml [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+    description: Presidio CLI version
+    required: false
+    default: "0.0.4"
+  lang-models:
+    description: List of additional lang models to install
+    require: false
+    default: ""
+  branding:
+    icon: 'alert-circle'
+    color: 'yellow'
```
Comment (cicdguy) [inline action.yaml]: `branding` should not be under `input`

File: action.yaml [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+        echo "install required models "
+        # python -m spacy download en_core_web_sm
+        python -m spacy download en_core_web_lg
+        echo "install additional models"
+        while IFS= read -r lang_model
+        do
+          if [[ ! -z "${lang_model}" ]]; then
+            python -m spacy download "${lang_model}"
+          fi
+        done <<< "${{ inputs.lang-models }}"
```
Comment (cicdguy) [inline action.yaml:77]: Nice!

**Comment by cicdguy** [comment]
What are all the PR comments coming from the bot about? Probably needs a refactor so it makes sense to humans.

**Comment by tomszosz** [comment]
> What are all the PR comments coming from the bot about? Probably needs a refactor so it makes sense to humans.

Those are examples from test execution with all supported output formats ( details in README - default type is `auto`)

**Review by KlaudiaBB — APPROVED** [review]

File: action.yaml (hunk 16 above)
Comment (tomszosz) [reply action.yaml]: thanks for finding

**Pull request merged**